### ISSUE TITLE
Orphan instances

### ISSUE THREAD

**Issue by MaxGabriel** [pbrisbin/bugsnag-haskell#48]
This isn't causing any particular problems for me, but I noticed the other day that the library declares two orphan `ToJSON` instances

```haskell
instance ToJSON ByteString where
    toJSON = String . decodeUtf8

instance ToJSON a => ToJSON (CI a) where
    toJSON = toJSON . CI.original
```

I'm not overly opposed to orphans, but the ByteString one seems really presumptuous given that a ByteString could be a JPEG or non UTF-8 text or something. It looks like the orphans are used for e.g. the `BugsnagRequest`, do you think those might be better as IP (from Data.IP) and Text? Or a newtype wrapper around `ByteString`?

```
 data BugsnagRequest = BugsnagRequest
    { brClientIp :: Maybe ByteString
    , brHeaders :: Maybe RequestHeaders
    , brHttpMethod :: Maybe Method
    , brUrl :: Maybe ByteString
    , brReferer :: Maybe ByteString
    }
```

**Comment by pbrisbin** [comment]
Yup, I don't disgree. This was just a short-cut taken because of the early nature of the library -- with the intention of cleaning up later.

I'm not sure when I'll have the time, so if you want to make a PR please feel free. I'm not super opinionated about how the orphans are removed (conversion to `Text`, `newtype`s, whatever), so feel free to make your own choices.

**Comment by pbrisbin** [comment]
Closed by #73 and released in v0.0.4.4.
